src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/UpdateAzureMLWebService.cs

```diff
@@ -0,0 +1,195 @@
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access keys for the web service.")]
+        public WebServiceKeys Keys { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access key for the storage account associated with the web service. This allows for key rotation if needed.")]
+        public string StorageAccountKey { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New diagnostics settings for the web service.")]
+        public DiagnosticsConfiguration Diagnostics { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New realtime endpoint runtime settings for the web service.")]
+        public RealtimeConfiguration RealtimeConfiguration { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated assets for the web service.")]
+        public IDictionary<string, AssetItem> Assets { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated input schema for the web service.")]
+        public ServiceInputOutputSpecification Input { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated output schema for the web service.")]
+        public ServiceInputOutputSpecification Output { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated global parameter values definition for the web service.")]
+        public IDictionary<string, string> Parameters { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated graph package for the web service.")]
+        public GraphPackage Package { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromObjectParameterSet, 
+            Mandatory = true, 
+            HelpMessage = "An updated definition object to update the referenced web service with.", 
+            ValueFromPipeline = true)]
+        public WebService ServiceUpdates { get; set; }
```

Comment:
That is the point.  Is it a better experience to be able to pipe an existing object and change some parameters in the cmdlet through changoign the parameter sets?  Right now the user doesn't even know which properties (other than name and resource group) are required.